Generate --all a different way

Insti opened the pull request:
Testing pain tends to be a sign that one's abstractions may be sub-optimal.

I tried a different approach by returning and iterating over an array of generators rather than calling something that also needed to construct them.

I think it turned out quite well.

What do you think?

Ref: https://github.com/exercism/xruby/pull/620

Insti committed:
Revert "add --all option to generator"

This reverts commit 67a8e995a7c90983cbaea730564934687ffdc2d7.

Insti committed:
Return an array of generators instead of just one.

Insti committed:
Pass exercise name as an argument to repository.

Insti committed:
Iterate over exercise names.

Use all exercise names if --all was specified, but most of the time
there will only be one.

Insti committed:
Remove obsolete CommandLine tests

Insti committed:
Add tests for ComandLine

100% coverage of CommandLine
`parse` is the only public method.

Insti commented:
The revert at the start makes the diff look a bit noisy, try just viewing all but the first commit.

kotp commented on lib/generator/command_line.rb:
```diff
@@ -11,13 +11,13 @@ def initialize(paths)
       parser = GeneratorOptparser.new(args, @paths)
       @options = parser.options
-      generator if parser.options_valid?
+      return generators if parser.options_valid?
```
I realize why the return is here, but with the plural form, this is clear even without the return explicitly there.

Insti committed:
Refactoring: Extract meta_generator_path method

Insti committed:
Refactor: Rename 'load_filename' to 'source_filename'

This method returns the filename of a `*_cases.rb` file.
I kept getting confused because it wasn't "load"ing anything.

`sourcefile_name` would aso be an option, but `filename` seems to be the
project standard for file names.

Insti committed:
Refactor: Remove obsolete comment.

Insti commented:
Refactoring of generator cases should probably be a separate PR but I am ok with sneaking it into this one if you don't feel like pulling them apart.
(Although it's just these 3 +  'clean up #available', so it should be easy to separate them.)

hilary commented on lib/generator/template_values.rb:
```diff
@@ -45,7 +45,7 @@ def extractor
 
     def cases_load_name
-      Files::GeneratorCases.load_filename(paths.track, exercise_name)
+      Files::GeneratorCases.source_filename(paths.track, exercise_name)
```
What is this change doing here? This isn't part of the changes needed.

Insti commented:
Closing old PR.
